File: module/lambda/codes/confirm_signup/confirm_signup.py
import boto3
import hmac
import hashlib
import base64
import os
from utils import logger, resp, db
POOL_ID = os.environ["POOL_ID"]
CLIENT_ID = os.environ["CLIENT_ID"]
CLIENT_SECRET = os.environ["CLIENT_SECRET"]
client = boto3.client('cognito-idp')

def get_secret_hash(username, client_id, client_secret):
    msg = username + client_id
    dig = hmac.new(
        str(client_secret).encode('utf-8'),
        msg=str(msg).encode('utf-8'),
        digestmod=hashlib.sha256
    ).digest()
    d2 = base64.b64encode(dig).decode()
    return d2


def lambda_handler(event, context):
    print(event)
    try:
        
        body = event['body'] if 'body' in event and event['body'] is not None else event
        user_name = body["email"]
        code = body["code"]
        
        
        secret_hash = get_secret_hash(user_name, CLIENT_ID, CLIENT_SECRET)
        response = client.confirm_sign_up(
            ClientId=CLIENT_ID,
            SecretHash=secret_hash,
            Username=user_name,
            ConfirmationCode=code,
            ForceAliasCreation=False
        )
    
        # if response:
        #     return "account confirmed successfully"
        status_code = 200
        resp["error"] = False
        resp["success"] = True
        resp["message"] = "account confirmed successfully"
    except Exception as e:
        status_code = 500
        logger.error(e)
        resp["message"] = "something went wrong"
        resp["error"] = True
        resp["success"] = False
    return callback_response(status_code)
    

def callback_response(code):
  return {
         'statusCode': code,
         'body': resp
     }

File: module/lambda/codes/confirm_signup/confirm_signup.js
// file: lambdaConfirmSignup.js
import crypto from "crypto";
import {
  CognitoIdentityProviderClient,
  ConfirmSignUpCommand,
} from "@aws-sdk/client-cognito-identity-provider";

const REGION = process.env.AWS_REGION;
const CLIENT_ID = process.env.CLIENT_ID;
const CLIENT_SECRET = process.env.CLIENT_SECRET;

const cognitoClient = new CognitoIdentityProviderClient({ region: REGION });

/**
 * Generate Cognito Secret Hash
 */
function getSecretHash(username, clientId, clientSecret) {
  const msg = username + clientId;
  const hmac = crypto.createHmac("sha256", clientSecret);
  hmac.update(msg);
  return hmac.digest("base64");
}

/**
 * Confirm signup Lambda handler
 */
export const handler = async (event) => {
  console.log("Event:", event);

  const resp = { error: false, success: false, message: "" };
  let statusCode = 200;

  try {
    const body =
      event.body && typeof event.body === "string"
        ? JSON.parse(event.body)
        : event.body || event;

    const username = body.email;
    const code = body.code;

    const secretHash = getSecretHash(username, CLIENT_ID, CLIENT_SECRET);

    const command = new ConfirmSignUpCommand({
      ClientId: CLIENT_ID,
      SecretHash: secretHash,
      Username: username,
      ConfirmationCode: code,
      ForceAliasCreation: false,
    });

    await cognitoClient.send(command);

    resp.success = true;
    resp.message = "Account confirmed successfully";
  } catch (err) {
    console.error("Confirm signup error:", err);
    statusCode = 500;
    resp.error = true;
    resp.success = false;
    resp.message = "Something went wrong";
  }

  return {
    statusCode,
    body: JSON.stringify(resp),
  };
};
